bundles/org.openhab.binding.heos/src/main/java/org/openhab/binding/heos/internal/discovery/HeosPlayerDiscovery.java

```diff
@@ -107,23 +105,24 @@ protected void startScan() {
                 logger.debug("Found: {} new Groups", groupMap.size());
                 ThingUID bridgeUID = bridge.getThing().getUID();
 
-                for (String groupGID : groupMap.keySet()) {
-                    HeosGroup group = groupMap.get(groupGID);
-                    String groupMemberHash = group.getGroupMemberHash();
+                for (Map.Entry<String, Group> entry : groupMap.entrySet()) {
+                    Group group = entry.getValue();
+                    String groupMemberHash = entry.getKey();
                     // Using an unsigned hashCode from the group members to identify
                     // the group and generates the Thing UID.
                     // This allows identifying the group even if the sorting within the group has changed
                     ThingUID uid = new ThingUID(THING_TYPE_GROUP, groupMemberHash);
                     Map<String, Object> properties = new HashMap<>();
-                    properties.put(PROP_NAME, group.getName());
-                    properties.put(PROP_GID, group.getGid());
-                    properties.put(PROP_GROUP_MEMBERS, group.getGroupMembersAsString());
-                    properties.put(PROP_GROUP_LEADER, group.getLeader());
-                    properties.put(PROP_GROUP_HASH, group.getGroupMemberHash());
-                    DiscoveryResult result = DiscoveryResultBuilder.create(uid).withLabel(group.getName())
+                    properties.put(PROP_NAME, group.name);
+                    properties.put(PROP_GID, group.id);
+                    String groupMembers = group.players.stream().map(p -> p.id).collect(Collectors.joining(";"));
+                    properties.put(PROP_GROUP_MEMBERS, groupMembers);
+                    properties.put(PROP_GROUP_LEADER, group.players.get(0).id);
+                    properties.put(PROP_GROUP_HASH, groupMemberHash);
+                    DiscoveryResult result = DiscoveryResultBuilder.create(uid).withLabel(group.name)
```

Comment:
The result at the player is build with 'withRepresentationProperty'. I'm not sure what exactly it is good for. But do we need this also at the groups?

Reply:
It does a deduplication based on the field with that name. We could consider using the hash here. It would cause a groep to be ignored when discovered if it already exists as such.

So hash feels good.

Reply:
See also: https://www.openhab.org/docs/developer/bindings/thing-xml.html#representation-property